File: payroll-api/src/main/java/com/company/payroll/service/SalaryService.java
package com.company.payroll.service;

import java.util.Optional;

import com.company.payroll.model.Salary;
import com.github.pagehelper.PageInfo;

public interface SalaryService {
	
	/**
	 * 
	 * @param sid
	 * @return
	 */
	Optional<Salary> findById(int sid);
	
	/**
	 * 
	 * @return
	 */
	PageInfo<Salary> list(int page, int offset);
	
	/**
	 * 
	 * @param salary
	 * @return
	 */
	Salary update(Salary salary);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/service/SalaryService.kt
package com.company.payroll.service

import com.company.payroll.model.StaffSalary

import com.github.pagehelper.PageInfo

interface SalaryService {
  /**
   *
   * @param sid
   * @return
   */
  fun findById(sid: Int): StaffSalary?

  /**
   *
   * @return
   */
  fun list(page: Int, offset: Int): PageInfo<StaffSalary>

  /**
   *
   * @param staffSalary
   * @return
   */
  fun update(staffSalary: StaffSalary): StaffSalary
}